whisper.cpp

```diff
@@ -1659,25 +1659,19 @@ static struct ggml_cgraph * whisper_build_graph_conv(
         ggml_set_name(cur, "embd_conv");
         wstate.embd_conv = cur;
     } else {
-#ifdef WHISPER_USE_COREML
-        cur = ggml_new_tensor_2d(ctx0, GGML_TYPE_F32, n_state, n_ctx);
-        ggml_allocr_alloc(alloc, cur);
+        // keep the "mel" tensor alive - we will use it to store the input data for the external encoders
+        // TODO: is there a better way to do this
+        mel = ggml_scale(ctx0, mel, 1.0f);
+        ggml_build_forward_expand(gf, mel);
 
-        if (!ggml_allocr_is_measure(alloc)) {
-            whisper_coreml_encode(wstate.ctx_coreml, mel->ne[0], mel->ne[1], (float *) mel->data, (float *) cur->data);
-        }
-#endif
-#ifdef WHISPER_USE_OPENVINO
         cur = ggml_new_tensor_2d(ctx0, GGML_TYPE_F32, n_state, n_ctx);
-        ggml_allocr_alloc(alloc, cur);
-
-        if (!ggml_allocr_is_measure(alloc)) {
-            whisper_openvino_encode(wstate.ctx_openvino, mel, cur);
-        }
-#endif
-
         ggml_set_name(cur, "embd_enc");
+        ggml_set_output(cur);
         wstate.embd_enc = cur;
+
+        // TODO: without this op, the "embd_enc" tensor ends up being not allocated
+        //       is there a better fix?
+        cur = ggml_scale(ctx0, cur, 1.0f);
```

Comment:
Only the tensors used in the nodes of the graph are allocated, but I don't think this is new. Did it work before alloc v3?

Reply:
I guess before it worked because of the explicit `ggml_allocr_alloc` calls